xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -496,11 +496,9 @@ void CGUIDialogVideoInfo::OnSearch(std::string& strSearch)
       if (iItem < 0)
         return;
 
-      CFileItem* pSelItem = new CFileItem(*items[iItem]);
+      CFileItemPtr pSelItem(new CFileItem(*items[iItem]));
 
-      OnSearchItemFound(pSelItem);
-
-      delete pSelItem;
+      OnSearchItemFound(pSelItem.get());
```

Comment:
Here, you create a new CFileItem on the heap with `new`. You then pass it to OnSearchItemFound() and destroy the CFileItem after (when the smart pointer falls out of scope).

However, even though OnSearchItemFound() takes a pointer, it doesn't care if the CFileItem is on the heap or the stack. Creating the CFileItem on the stack is faster because LIFO is O(1). You you could do this:

```c++
CFileItem selItem(*items[iItem]);
OnSearchItemFound(&selItem);
```